### PR TITLE
special_role/objective refactor

### PR DESCRIPTION
**Pull request by tigercat2000**
Refactored most instances of special_role setting/checking to use new
SPECIAL_ROLE_X defines in code/__DEFINES/gamemode.dm

Removed relative pathing from objective.dm and split large if statements
into multiple, plus a bunch of styling fixes

Refactored every instance of `istype(ticker.mode` into a GAMEMODE_IS_X
define in code/__DEFINES/gamemode.dm. Done primarily for when someone gets
around to making gamemode code less AWFUL

### CODE REVIEW

**Comment by RemieRichards** [comment]
Why not just use the ```ROLE_X``` defines?

**Comment by tigercat2000** [comment]
@RemieRichards No one was really sure if it was alright to use those outside of whatever it is they are used for now. It'd be pretty easy to mass replace these, though.

**Comment by RemieRichards** [comment]
@tigercat2000 they're just text defines, they're good for reuse.

**Comment by Fox-McCloud** [comment]
Yeah, should definitely be using the `ROLE_X` defines for this. Just having the defines doesn't add them to the menu or anything like that--that's handled elsewhere.

**Comment by tigercat2000** [comment]
I actually did find one really good reason _not_ to use the `ROLE_X` defines.

Most `ROLE_X` defines are used for jobbans; This provides a problem, specifically, roles such as abductors, blobs, or xenomorphs. 
All three of them need separate special_role defines for all of their different portions- abductors have the agent and scientist, where special_role **really** matters, blobs have a role for the person about to burst and when they have actually become the blob, and xenomorphs have a different role for every class. 
However, all of them also should have only one job-ban entry attached. It'd lead to confusing defines if they were inconsistent like having a general `abductor` role as well as an `abductor scientist` and `abductor agent` role.

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -16,329 +16,290 @@ var/list/potential_theft_objectives=subtypesof(/datum/theft_objective) \
...
+/datum/objective/maroon/check_completion()
+	if(target && target.current)
+		if(target.current.stat == DEAD)
+			return 1
+		if(issilicon(target.current))
+			return 1
+		if(isbrain(target.current))
+			return 1
+		// TODO: Tie into space manager
+		if(target.current.z in config.admin_levels)
```
Comment (KasparoVy) [inline code/game/gamemodes/objective.dm]: Ack, sorry about the conflicts. Due to the way the assassinate check_completion is structured in this PR the changes I made in #5346 to the proc are irrelevant-- but this here undoes [this line](https://github.com/KasparoVy/Paradise/blob/6edee2b55a6feb580a882c96f6b10b1ca3a29f26/code/game/gamemodes/objective.dm#L180) and [this line](https://github.com/KasparoVy/Paradise/blob/6edee2b55a6feb580a882c96f6b10b1ca3a29f26/code/game/gamemodes/objective.dm#L181) from #5346

**Pull request merged**